common/OpTestUtil.py

```diff
@@ -1913,6 +1913,54 @@ def wait_for(self, func, timeout, first=0.0, step=1.0, text=None, args=None, kwa
             time.sleep(step)
         return None
 
+    def gather_os_logs(self, list_of_files=[], list_of_commands=[], output_dirname=None):
+        host = self.conf.host()
+        if not output_dirname:
+            output_dirname = "Logs_%s" % (time.asctime(time.localtime())).replace(" ", "_")
+        output_dir = os.path.join(host.results_dir, output_dirname)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        self.gather_files_log(list_of_files, output_dir)
+        self.gather_command_log(list_of_commands, output_dir)
+
+    def gather_files_log(self, list_of_files=[], output_dir=None):
+        if not output_dir:
+            raise OpTestError("Provide path to the output directory (output_dir).")
+        host = self.conf.host()
+
+        default_files = ['/var/log/messages', '/var/log/boot.log']
+        list_of_files.extend(default_files)
+
+        try:
+            host.host_run_command("mkdir -p file-logs")
+            for file in set(list_of_files):
+                fn = "%s.log" % '-'.join(file.strip(os.path.sep).split(os.path.sep))
+                host.host_run_command("cp %s file-logs/%s" % (file, fn))
+            host.copy_files_from_host(sourcepath=output_dir, destpath="file-logs")   
+            host.host_run_command("rm -rf file-logs")
+            return True
+        except CommandFailed as cf:
+            raise cf
+
+    def gather_command_log(self, list_of_commands=[], output_dir=None):
+        if not output_dir:
+            raise OpTestError("Provide path to the output directory (output_dir).")
+        host = self.conf.host()
+
+        default_commands = ['dmesg', 'journalctl -a']
+        list_of_commands.extend(default_commands)
+
+        try:
+            host.host_run_command("mkdir -p cmd-logs")
+            for cmd in set(list_of_commands):
+                fn = "%s.log" % '-'.join((re.sub(r'[^a-zA-Z0-9]', ' ', cmd)).split())
+                host.host_run_command("%s > cmd-logs/%s" % (cmd, fn))
+            host.copy_files_from_host(sourcepath=output_dir, destpath="cmd-logs")   
+            host.host_run_command("rm -rf cmd-logs")
+            return True
+        except CommandFailed as cf:
+            raise cf
```

Comment:
variable name length please, i know we are still using op-test having this but please consider this